From 20317bdfb46942ee8e65e8bc8ae8fd2a637447d2 Mon Sep 17 00:00:00 2001
From: "Xiang, Haihao" <haihao.xiang@intel.com>
Date: Mon, 2 Feb 2015 16:16:53 +0800
Subject: [PATCH] Disable upper bound check for decoding on BDW+

We saw pixel corruption in stress test with the wrong up-bound address.
For simplicity, disable upper bound check in the driver.

Signed-off-by: Xiang, Haihao <haihao.xiang@intel.com>
Reviewed-by: Sean V Kelley <sean.v.kelley@intel.com>
Tested-by: Zhao Yakui <yakui.zhao@intel.com>
---
 src/gen8_mfd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gen8_mfd.c b/src/gen8_mfd.c
index a080504..2fabb3b 100644
--- a/src/gen8_mfd.c
+++ b/src/gen8_mfd.c
@@ -281,7 +281,7 @@ gen8_mfd_ind_obj_base_addr_state(VADriverContextP ctx,
     OUT_BCS_BATCH(batch, 0);
     OUT_BCS_BATCH(batch, 0);
 	/* Upper bound 4-5 */	
-    OUT_BCS_BATCH(batch, 0x80000000); /* must set, up to 2G */
+    OUT_BCS_BATCH(batch, 0);
     OUT_BCS_BATCH(batch, 0);
 
 	/* MFX indirect MV 6-10 */
@@ -2509,7 +2509,7 @@ gen8_jpeg_wa_ind_obj_base_addr_state(VADriverContextP ctx,
                   gen7_mfd_context->jpeg_wa_slice_data_bo,
                   I915_GEM_DOMAIN_INSTRUCTION, 0,
                   0);
-    OUT_BCS_BATCH(batch, 0x80000000); /* must set, up to 2G */
+    OUT_BCS_BATCH(batch, 0);
     OUT_BCS_BATCH(batch, 0); /* ignore for VLD mode */
     OUT_BCS_BATCH(batch, 0);
     OUT_BCS_BATCH(batch, 0); /* ignore for VLD mode */
-- 
2.1.0

